src/EnergyPlus/Boilers.hh

```diff
@@ -68,6 +68,10 @@ namespace EnergyPlus {
 // Forward declarations
 struct EnergyPlusData;
 
+namespace Curve {
```

Comment:
Using forward declarations instead of including `CurveManager.hh` in the header file. Should we start using `X.fwd.hh` files or are ad hoc forward declarations like this okay?

Reply:
I tend to think it's better to stick with the bare forward declarations. The downside is that curves are in a lot of headers, but the `X.fwd.hh` approach would lead to even more headers and we already have a lot of files.

Reply:
> I tend to think it's better to stick with the bare forward declarations. The downside is that curves are in a lot of headers, but the `X.fwd.hh` approach would lead to even more headers and we already have a lot of files.

What if we have a single `Common.fwd.hh` file for `Schedule`, `Curve`, `Glycol`, `Refrig`, etc.

Reply:
I guess I'd still rather just do the in situ forward declaration and only forward declare what gets used. I feel like we'd end up with half of the objects forward declared, but I could be wrong. Do we know if forward declarations will still be a thing when modules replace inclusion? I know that's likely in the grim future of C++40k and all, but I haven't kept track of that stuff.

Reply:
I certainly do not keep track of such things. 

Forward declarations are very cheap. It's just a name in a lookup table that says "I am allowed to declare pointers to objects of this name." There is basically no cost to having a thousand of them. 

The benefit of forward declarations is more for incremental compilation than for full compilation, because it allows you to make changes to `.hh` files and have those changes trigger recompilation in a smaller number of other `.hh` files in which they are included. I will leave this as is for now. Thanks for the discussion.